engine/src/render/cascaded_shadow_maps/plugin.cpp

```diff
@@ -0,0 +1,82 @@
+#include <cubos/core/io/window.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+
+#include <cubos/engine/render/camera/camera.hpp>
+#include <cubos/engine/render/camera/plugin.hpp>
+#include <cubos/engine/render/cascaded_shadow_maps/plugin.hpp>
+#include <cubos/engine/render/shadows/directional_caster.hpp>
+#include <cubos/engine/render/shadows/plugin.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+using cubos::core::io::Window;
+
+CUBOS_DEFINE_TAG(cubos::engine::createCascadedShadowMapsTag);
+CUBOS_DEFINE_TAG(cubos::engine::drawToCascadedShadowMapsTag);
+
+void cubos::engine::cascadedShadowMapsPlugin(Cubos& cubos)
+{
+    cubos.depends(cameraPlugin);
+    cubos.depends(shadowsPlugin);
+    cubos.depends(windowPlugin);
+
+    cubos.tag(createCascadedShadowMapsTag);
+    cubos.tag(drawToCascadedShadowMapsTag).after(createCascadedShadowMapsTag);
+
+    cubos.system("create cascaded shadow maps")
+        .tagged(createCascadedShadowMapsTag)
+        .call([](const Window& window, Query<DirectionalShadowCaster&> query, Query<Entity, const Camera&> cameras) {
+            for (auto [caster] : query)
+            {
+                // Remove shadow maps for cameras that no longer exist
+                std::vector<Entity> removedCameras;
+                for (auto& [cameraEntity, shadowMap] : caster.shadowMaps)
+                {
+                    if (!cameras.at(cameraEntity))
+                    {
+                        removedCameras.push_back(cameraEntity);
+                    }
+                }
+                for (auto& cameraEntity : removedCameras)
+                {
+                    caster.shadowMaps.erase(cameraEntity);
+                }
+
+                // Recreate shadow maps if the size or number of splits changed
+                if (caster.size != caster.configSize || caster.numSplits != caster.configNumSplits)
+                {
+                    caster.size = caster.configSize;
+                    caster.numSplits = caster.configNumSplits;
+
+                    for (auto& [camera, shadowMap] : caster.shadowMaps)
+                    {
+                        shadowMap.resize(window->renderDevice(), caster.size, caster.numSplits);
+                    }
+                }
+
+                // Set split distances
+                if (caster.configSplitDistances.size() == static_cast<unsigned long>(caster.numSplits))
+                {
+                    caster.splitDistances = caster.configSplitDistances;
+                }
+                else
+                {
+                    // Set default split distances
+                    caster.splitDistances.clear();
+                    for (int i = 0; i < caster.numSplits; i++)
+                    {
+                        caster.splitDistances.push_back(float(i) / float(caster.numSplits));
+                    }
+                }
+
```

Comment:
You could check here if the `numSplits`/`splitDistances.size()` is > MaxSplits, and in that case, issue a warning and clamp/crop it down to the allowed number.